bayesian/auth.py

```diff
@@ -26,3 +28,41 @@ def get_access_token(service_name):
 
     except Exception:
         logger.error('Unable to connect to Auth service')
+
+
+def validate_user(view):
+    """Validate and get user type based on UUID from the request."""
+    @wraps(view)
+    def wrapper(*args, **kwargs):
+        """Read uuid and decides user type based on its validity."""
+        # Rule of UUID validation and setting user status ::
+        #  ==============================================================
+        #   UUID in request | UUID in RDS | RDS User State | User Status
+        #  ==============================================================
+        #    MISSING        | -- NA --    | -- NA --       | FREE
+        #    PRESENT        | MISSING     | -- NA --       | FREE
+        #    PRESENT        | PRESENT     | REGISTERED     | REGISTERED
+        #    PRESENT        | PRESENT     | !REGISTERED    | FREE
+        #  ==============================================================
+
+        # By default set this to 'freetier'.
+        g.registration_status = UserStatus.FREETIER
+
+        # Read uuid from request header.
+        uuid = request.headers.get('uuid', None)
+
+        # Check user with uuid if it is present in RDS.
+        if uuid is not None:
+            try:
+                # Read user details from RDS based on uuid
+                user = get_user(uuid)
+            except UserException as e:
+                logger.warning("Unable to get status for uuid=%s, err=%s", uuid, e)
+            else:
+                if user.status == 'REGISTERED':
+                    g.user_status = UserStatus.REGISTERED
```

Comment:
When dealing with enum, you can use a simple approach to map to values.

```suggestion
                g.user_status = UserStatus[user.status]
```